testutils/base.go

```diff
@@ -24,22 +24,20 @@ import (
 	"github.com/cockroachdb/cockroach/security"
 )
 
-// NewTestBaseContext creates a base context for testing.
+// NewRootTestBaseContext creates a base context for testing.
 // This uses embedded certs and the "root" user (default client user).
-// The certs file loader is overriden in individual main_test files.
-// This is meant to be used by external clients (as opposed to nodes).
-func NewTestBaseContext() *base.Context {
+// The "root" user has client certificates only.
+func NewRootTestBaseContext() *base.Context {
```

Comment:
Changing to both calling a single method and passing the user.
However, changing them to vars is dangerous as there are quite a few tests that grab a test context and modify an particular field without resetting it later.
Is there a general position for or against struct pointers in general?